### ISSUE TITLE
Don't work with UserNames

### ISSUE THREAD

**Issue by kodoor** [rmcrackan/Libation#9]
I am a long time user who uses Audible with a username instead of an Amazon Account Email address.
Any way to use it with user name instead?
At the Web login prompt in a browser you get the choice of using an Amazon account or a User name.

**Comment by rmcrackan** [comment]
tl;dr: no. It's not possible

The long answer:

This kills me every time this comes up; I really want to support older users like you. The login in Libation is a product of hacking around and observing what happens when I log in. Since I can't do that with non-amazon audible usernames, there's no way for me to support your style of login. (note: I'm not asking for permissions to your login and would never in a million years feel comfortable with that access.)

There are a few people like me who have reverse engineered audible's communications and we're all stuck on the wrong side of the cat and mouse game. Unfortunately, amazon/audible refuses to give official supported access to programmers so we can make software to support audible libraries. (It's possible that such investigations are breaking their terms and conditions.) We have to go through complex investigations which could break overnight if audible decides to do things a little differently.

**Comment by mkb79** [comment]
Hi. Maybe it should possible. Login with a amazon account refers to https://www.amazon.{domain}/ap/signin site. If you login with an old audible account you have to login at https://www.audible.{domain}/ap/signin. The params are the same except pageId and openid.assoc_handle who have an additional _lap_ in string. 

You can try to change the login url to https://www.audible.{domain}/ap/signin.

**Comment by rmcrackan** [comment]
@mkb79 

> The params are the same except pageId and openid.assoc_handle who have an additional _lap_ in string

I'm not sure what you mean by this. Have you gotten it working in your project?

Since this is your area of expertise, what I probably need to do is stop my efforts to translate python => c# and instead find a way to interop with your package so we can each focus on what we do best. dot-net/python interop is simple in concept but difficult for deployment because of dependencies. I may look into this soon.

**Comment by mkb79** [comment]
I mean login with an amazon accounts forwards you to www.amazon.{domain}/ap/signin and login with old audible accounts forwards to www.audible.{domain}/ap/signin. 

The url query of the login page GET request seams the same in booth cases except the things I wrote above. 

I don’t have an old audible account to verify my idea. But it could be possible to change login path to audible and then login with an old audible account.

**Comment by rmcrackan** [comment]
Thanks, I'll look into this. Do you support it in your py library?

**Comment by rmcrackan** [comment]
> I am a long time user who uses Audible with a username instead of an Amazon Account Email address.
> Any way to use it with user name instead?
> At the Web login prompt in a browser you get the choice of using an Amazon account or a User name.

I have some ideas worth trying. Frustratingly, github has no private message feature. On reddit can you PM me your email and we can talk back and forth more freely?

**Comment by mkb79** [comment]
I don’t support authenticate with username because I can’t test it.
